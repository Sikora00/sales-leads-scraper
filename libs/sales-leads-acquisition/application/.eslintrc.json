{
  "extends": ["../../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": ["libs/sales-leads-acquisition/application/tsconfig.*?.json"]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
