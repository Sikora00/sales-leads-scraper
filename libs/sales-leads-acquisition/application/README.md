# sales-leads-acquisition-application

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test sales-leads-acquisition-application` to execute the unit tests via [Jest](https://jestjs.io).
