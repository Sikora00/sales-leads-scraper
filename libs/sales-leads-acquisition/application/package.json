{
  "name": "@sales-leads/sales-leads-acquisition/application",
  "version": "0.0.1"
}
