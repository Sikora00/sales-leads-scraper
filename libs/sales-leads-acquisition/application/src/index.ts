export * from './lib/sales-leads-acquisition-application.module';
export * from './lib/sales-leads-acquisition.facade';
export * from './lib/commands/find-new-sales-leads/find-new-sales-leads.command';
export * from './lib/ports/sales-leads-finder';
export * from './lib/ports/sales-lead.repository';
