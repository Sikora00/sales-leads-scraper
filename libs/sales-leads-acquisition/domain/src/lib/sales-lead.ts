import { AggregateRoot } from '@nestjs/cqrs';
import { Url } from '@sales-leads/shared/domain-technical';
import { Country } from './value-objects/country';
import { Industry } from './value-objects/industry';
import { CompanySize } from './value-objects/company-size';
import { SourceAdvertisement } from './value-objects/source-advertisement';
import { City } from './value-objects/city';
import { NewSalesLeadAcquiredEvent } from './events/new-sales-lead-acquired.event';

export class SalesLead extends AggregateRoot {
  private city: City;
  private companySize: CompanySize;
  private companyWebsite: Url;
  private country: Country;
  private industry: Industry;
  private name: string;
  private sourceAdvertisement: SourceAdvertisement;

  private constructor(data: SalesLeadData) {
    super();
    this.city = data.city;
    this.companySize = data.companySize;
    this.companyWebsite = data.companyWebsite;
    this.country = data.country;
    this.industry = data.industry;
    this.name = data.name;
    this.sourceAdvertisement = data.sourceAdvertisement;
  }

  static acquired(data: SalesLeadData): SalesLead {
    const salesLead = new SalesLead(data);
    salesLead.apply(new NewSalesLeadAcquiredEvent());
    return salesLead;
  }

  static restore(data: SalesLeadData): SalesLead {
    return new SalesLead(data);
  }

  getCity(): City {
    return this.city;
  }

  getCompanySize(): CompanySize {
    return this.companySize;
  }

  getCompanyWebsite(): Url {
    return this.companyWebsite;
  }

  getCountry(): Country {
    return this.country;
  }

  getIndustry(): Industry {
    return this.industry;
  }

  getName(): string {
    return this.name;
  }

  getSourceAdvertisement(): SourceAdvertisement {
    return this.sourceAdvertisement;
  }
}

export interface SalesLeadData {
  city: City;
  companySize: CompanySize;
  companyWebsite: Url;
  country: Country;
  industry: Industry;
  name: string;
  sourceAdvertisement: SourceAdvertisement;
}
