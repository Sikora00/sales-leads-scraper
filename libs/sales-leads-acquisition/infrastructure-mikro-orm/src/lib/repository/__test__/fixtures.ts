import { Test } from '@nestjs/testing';
import { MikroOrmModule } from '@mikro-orm/nestjs';
import { SalesLeadsAcquisitionInfrastructureMikroOrmModule } from '../../sales-leads-acquisition-infrastructure-mikro-orm.module';
import {
  City,
  CompanySize,
  Country,
  Industry,
  SalesLead,
  SourceAdvertisement,
} from '@sales-leads/sales-leads-acquisition/domain';
import { Url } from '@sales-leads/shared/domain-technical';
import { INestApplication } from '@nestjs/common';
import { MikroORM } from '@mikro-orm/core';
import { join } from 'path';

export class Fixtures {
  static async getSandbox(): Promise<INestApplication> {
    const module = Test.createTestingModule({
      imports: [
        MikroOrmModule.forRoot({
          dbName: join(__dirname, 'db.sqlite'),
          type: 'sqlite',
          autoLoadEntities: true,
        }),
        SalesLeadsAcquisitionInfrastructureMikroOrmModule,
      ],
    });
    const sandbox = await module.compile();
    return sandbox.createNestApplication();
  }

  static async buildSchema(sandbox: INestApplication): Promise<void> {
    const orm = sandbox.get(MikroORM);
    const generator = orm.getSchemaGenerator();
    await generator.updateSchema();
  }

  static getSampleSalesLead(): SalesLead {
    return SalesLead.restore({
      city: new City('Warsaw'),
      companySize: new CompanySize('50-100'),
      companyWebsite: new Url('https://valueadd.pl'),
      country: new Country('PL'),
      industry: new Industry('IT'),
      name: 'ValueAdd',
      sourceAdvertisement: new SourceAdvertisement(
        new Url('https://valueadd.pl/career'),
        'NestJS Developer'
      ),
    });
  }

  static async dropDb(sandbox: INestApplication): Promise<void> {
    const orm = sandbox.get(MikroORM);
    const generator = orm.getSchemaGenerator();
    await generator.dropSchema();
  }
}
