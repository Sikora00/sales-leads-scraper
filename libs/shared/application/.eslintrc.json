{
  "extends": ["../../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": ["libs/shared/application/tsconfig.*?.json"]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
