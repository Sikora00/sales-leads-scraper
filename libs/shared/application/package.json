{
  "name": "@sales-leads/shared/application",
  "version": "0.0.1"
}
