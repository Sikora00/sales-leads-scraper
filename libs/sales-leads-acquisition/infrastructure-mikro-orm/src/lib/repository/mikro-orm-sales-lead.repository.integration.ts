import { Fixtures } from './__test__/fixtures';
import { SalesLeadRepository } from '@sales-leads/sales-leads-acquisition/application';
import { MikroORM, RequestContext } from '@mikro-orm/core';
import { SalesLeadEntity } from '../entities/sales-lead.entity';
import { SqliteDriver } from '@mikro-orm/sqlite';
import { INestApplication } from '@nestjs/common';

describe('MikroOrmSalesLeadRepository Integration', () => {
  let sandbox: INestApplication;
  beforeAll(async () => {
    sandbox = await Fixtures.getSandbox();
    await Fixtures.buildSchema(sandbox);
  });
  describe('when saving new sales lead', () => {
    beforeAll(async () => {
      return new Promise((resolve) => {
        RequestContext.create(sandbox.get(MikroORM).em, async () => {
          const repository = sandbox.get(SalesLeadRepository);
          await repository.save(Fixtures.getSampleSalesLead());
          resolve();
        });
      });
    });
    test('new record should be saved in the database', async () => {
      const orm: MikroORM<SqliteDriver> = sandbox.get(MikroORM);
      expect(await orm.em.find(SalesLeadEntity, {})).toMatchSnapshot();
    });
  });

  afterAll(async () => {
    await Fixtures.dropDb(sandbox);
    await sandbox.close();
  });
});
