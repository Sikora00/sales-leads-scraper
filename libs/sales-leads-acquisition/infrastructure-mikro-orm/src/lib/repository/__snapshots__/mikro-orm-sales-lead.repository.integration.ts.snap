// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MikroOrmSalesLeadRepository Integration when saving new sales lead new record should be saved in the database 1`] = `
Array [
  Object {
    "city": "Warsaw",
    "companySize": "50-100",
    "companyWebsite": "https://valueadd.pl/",
    "country": "PL",
    "id": "4ef28c8d-1ede-45e2-9deb-4f8a815c562d",
    "industry": "IT",
    "name": "ValueAdd",
    "sourceAdvertisementTitle": "NestJS Developer",
    "sourceAdvertisementUrl": "https://valueadd.pl/career",
  },
]
`;
